components/eam/src/chemistry/mozart/UCI_fjx_cmn_mod.f90

```diff
@@ -0,0 +1,262 @@
+!------------------------------------------------------------------------------
+!    'fjx_cmn_mod.f90'  for Solar/Cloud/Fast-J code v 7.6c (prather 06/2019)
+!------------------------------------------------------------------------------
+
+      MODULE FJX_CMN_MOD
+
+      implicit none
+      public
+
+!------------------------------------------------------------------------------
+      logical  LRRTMG, LCLIRAD, LGGLLNL  ! set in fjx_init_mod based on W_rrtmg
+      integer  NWBIN, NSBIN     ! dimensions of readin spec - NWBIN can zero out strat-wavels
+!-------------basic vertical grid----------these can be changed as needed, needed for dimensions
+      integer, parameter :: &  !  these must e set to exact atmospheric dimensions
+            LPAR = 72,      &  !  # layers in model (one layer is added to go to ZTOP+ZZHT)
+            LWEPAR = 72 !47    !  # layers that have clouds (LWEtPAR < LPAR)
```

Comment:
This makes the code not flexible when # of vertical levels changes.

Reply:
Yes.  However, vertical level changes are rare and this is a stealth feature.  There is also a check that will alert the user to fix the problem.   While this could be better integrated into E3SM, we also wanted to maintain as much similarity to the original Fast-J implementation as possible, so future improvements to Fast-J by the original developers (@pratherUCI ) can be merged in more easily and we don't end up in a backwater.

Reply:
Thanks for chiming in @cameronsmith1 . When I posted that comment, I didn't realize that was the cause of build failure for non-conventional physics cases like MMF1. We ended up having to change that. Just keep this for record. We may need to do the same when bringing in future versions of Fast-J.

Reply:
Hi @wlin7.  Was Fast-J turned on for MMF1?   If not, I cannot think why it caused a build failure, unless the compiler was being extra careful comparing procedure arguments inside and outside the call, even though they wouldn't be used.  

I'm glad you fixed the issue :-).